src/p2p/swarm.rs

```diff
@@ -222,17 +168,27 @@ impl NetworkBehaviour for SwarmApi {
     ) {
         // TODO: could be that the connection is not yet fully established at this point
         trace!("inject_connected {} {:?}", peer_id, cp);
-        let addr = connection_point_addr(cp);
+        let mut addr = connection_point_addr(cp).to_owned();
+
+        if !addr
+            .iter()
+            .any(|protocol| matches!(protocol, Protocol::P2p(_)))
+        {
```

Comment:
I think it's by design from rust-libp2p that this branch will always be taken.

I wonder if we should instead of future proofing against design changes just assert this? Other stuff will have changed if this changes I think?

Reply:
Not a high chance that we'll change it, but this can be `debug_asserted` I guess :+1:.